processor/src/operations/mod.rs

```diff
@@ -171,34 +171,51 @@ impl Process<super::MemAdviceProvider> {
 
     /// Instantiates a new blank process for testing purposes. The stack in the process is
     /// initialized with the provided values.
-    fn new_dummy(stack_inputs: &[u64]) -> Self {
-        let inputs = super::ProgramInputs::new(stack_inputs, &[], vec![]);
-        let mut process = Self::new(&Kernel::default(), inputs.unwrap());
+    fn new_dummy(stack: super::StackInputs) -> Self {
```

Comment:
What's the reason for changing the signature here and in other places? Seems like this leads to quite a few lines of extra code and makes testing a bit less convenient.

Reply:
The reason is to make the inputs of the test functions similar to the ones required by the API so we end up testing more of what we expose.

That isn't a strong reason tho and might be just `u64` as well.

Reply:
I don't feel very strongly, but deologically I agree with the pattern @vlopes11 is following. I like having the call to `Process::new_dummy` be similar to calling `Process::new`. I don't particularly see why we should be failing in `new_dummy` if the test inputs are incorrect, given that providing incorrect inputs in a non-test case would cause a failure before we instantiate the `Process`. So failing in the test feels correct, since essentially there was an error with the test "setup" phase (either that or we're explicitly testing bad inputs, in which case failing in the test also feels correct).

Reply:
OK - then let's leave it as is.